src/main/scala/org/apache/spark/ml/feature/RapidsPCA.scala

```diff
@@ -168,13 +171,34 @@ class RapidsPCAModel(
    */
   override def transform(dataset: Dataset[_]): DataFrame = {
     val outputSchema = transformSchema(dataset.schema, logging = true)
-
+    val cols_A = dataset.select($(inputCol)).first.size
+    class gpuTransform extends Function[Array[Double], Array[Double]] with RapidsUDF with Serializable {
+      override def evaluateColumnar(args: ColumnVector*): ColumnVector ={
+        require(args.length == 1, s"Unexpected argument count: ${args.length}")
+        val input = args.head
+        val rows_A = input.getRowCount.toInt
+        val AdevAddr = input.getData.getAddress
+        var C: Long = 0
+        // A: raw data, B: pc, C: output, deviceID= 0 for test
+        val lengthInBytes = RAPIDSML.gemm_test(RAPIDSML.CublasOperationT.CUBLAS_OP_N.id,
+          RAPIDSML.CublasOperationT.CUBLAS_OP_N.id,
+          rows_A, pc.numCols, cols_A, 1.0, AdevAddr, cols_A, pc, cols_A, 0.0, C, rows_A, 0)
+        val dmb = new DeviceMemoryBuffer(C, (rows_A*pc.numCols*8).toLong, Cuda.DEFAULT_STREAM)
```

Comment:
This constructor is accessible inside package `ai.rapids.cudf`.  So you need to put this code under that package.